src/SOS/lldbplugin/services.cpp

```diff
@@ -2229,14 +2264,72 @@ LLDBServices::GetThreadTeb(
     return E_NOTIMPL;
 }
 
-HRESULT LLDBServices::GetSymbolPath(
+HRESULT 
+LLDBServices::GetSymbolPath(
     PSTR buffer,
     ULONG bufferSize,
     PULONG pathSize)
 {
     return E_NOTIMPL;
 }
  
+HRESULT 
+LLDBServices::GetSymbolByOffset(
+    ULONG moduleIndex,
+    ULONG64 offset,
+    PSTR nameBuffer,
+    ULONG nameBufferSize,
+    PULONG nameSize,
+    PULONG64 displacement)
+{
+    return GetNameByOffset(moduleIndex, offset, nameBuffer, nameBufferSize, nameSize, displacement);
```

Comment:
Why do we need GetSymbolByOffset and GetNameByOffset

Reply:
GetNameByOffset is the original services code that was implementing the dbgeng function with the same name. It is still needed/called by the native SOS code.  GetSymbolByOffset has different semantics: search for the address in just the specified module instead of all modules. 

I have been trying not to change the ILLDBServices interface functions protocol types or semantics so the plugin could actually be used with an old libsos.so from the 2.1 runtime. ILLDBServices2 and IDebuggerServices can change still they are new.

Reply:
Mostly asking as it immediately just calls the other method with the same arguments. I guess the other one accepts the any token.